### PR TITLE
Deniz

### PR DESCRIPTION
**Pull request by dnzdlklc**
Change whitelist valve and added ability to increase per person cap after first 24 hours

### CODE REVIEW

File: contracts/Bounty0xCrowdsale.sol [hunk 1]
```diff
@@ -84,9 +87,13 @@ contract Bounty0xCrowdsale is Pausable, TokenController {
         uint256 contributionAmount = msg.value;
         require(contributionAmount > 0);
 
-        if (whitelistIsActive) {
-            require(bounty0xToken.balanceOf(msg.sender).add(contributionAmount) <= maximumParticipationAmount);
-            require(whitelistContributors[msg.sender]);
+        if (whitelistFilteringActive) {
+            if (now < WHITELIST_END_DATE) {
```
Comment (moodysalem) [inline contracts/Bounty0xCrowdsale.sol]: why do we need this and the boolean? better to have just one imo

File: contracts/Bounty0xCrowdsale.sol [hunk 2]
```diff
@@ -84,9 +87,13 @@ contract Bounty0xCrowdsale is Pausable, TokenController {
...
+                require(bounty0xToken.balanceOf(msg.sender).add(contributionAmount) <= maximumParticipationAmount);
```
Comment (moodysalem) [inline contracts/Bounty0xCrowdsale.sol]: maximum participation amount is in ether, contribution amount is in ether, we are adding the bounty0x token balance here

File: contracts/Bounty0xCrowdsale.sol [hunk 3]
```diff
@@ -14,6 +14,7 @@ contract Bounty0xCrowdsale is Pausable, TokenController {
     Bounty0xToken public bounty0xToken;                                 // Reward tokens to compensate in
     Bounty0xPresaleDistributor public presaleDistributor;               // contract that manages distributing presale awards
 
+    address public multiSigWallet                                       // Multi-sig wallet to transfer contributions to
```
Comment (moodysalem) [inline contracts/Bounty0xCrowdsale.sol]: we already have a bounty0xwallet variable

File: contracts/Bounty0xCrowdsale.sol [hunk 4]
```diff
@@ -84,9 +87,13 @@ contract Bounty0xCrowdsale is Pausable, TokenController {
...
 
-        if (whitelistIsActive) {
-            require(bounty0xToken.balanceOf(msg.sender).add(contributionAmount) <= maximumParticipationAmount);
-            require(whitelistContributors[msg.sender]);
+        if (whitelistFilteringActive) {
+            if (now < WHITELIST_END_DATE) {
+                require(bounty0xToken.balanceOf(msg.sender).add(contributionAmount) <= maximumParticipationAmount);
+                require(whitelistContributors[msg.sender]);
+            } else if (now > WHITELIST_END_DATE) {
+                increasePerCapAfterWhitelistPeriod();
```
Comment (moodysalem) [inline contracts/Bounty0xCrowdsale.sol]: imo contributors should not be wasting gas on this function call-remove the whitelist cap variable and just check if the whitelist contribution period is over

File: contracts/Bounty0xCrowdsale.sol [hunk 5]
```diff
@@ -182,6 +189,17 @@ contract Bounty0xCrowdsale is Pausable, TokenController {
         return true;
     }
 
+    // @notice Kill method should not really be needed, but just in case
+    function kill(address _to) onlyOwner external {
```
Comment (moodysalem) [inline contracts/Bounty0xCrowdsale.sol]: i don't think we should ever let this contract be killed, especially if it's the permanent token controller. my thought was that the token controller would be changed at the end of the crowdsale to a contract owned by the bounty0xwallet that allows for upgrading the token and stopping transfers

File: contracts/Bounty0xCrowdsale.sol [hunk 6]
```diff
@@ -182,6 +189,17 @@ contract Bounty0xCrowdsale is Pausable, TokenController {
...
+        selfDestruct(_to);
+    }
+
+    // Internal function to increase per person cap to $10k after first 24 hours of crowdsale
+    function increasePerCapAfterWhitelistPeriod() internal {
```
Comment (moodysalem) [inline contracts/Bounty0xCrowdsale.sol]: this can go away if we agree to just use WHITELIST_END_DATE

File: contracts/Bounty0xCrowdsale.sol (hunk 3 above)
Comment (dnzdlklc) [reply contracts/Bounty0xCrowdsale.sol]: 👍

**Pull request merged**